### DISCUSSION TITLE
Strategy for caching Docker images to be used with Docker run (not Docker build)

### DISCUSSION THREAD

**Discussion by AlexDCraig** [community/community#26643] [Actions]
<p>Does there exist any working examples for a process that would cache images repeatedly used to perform docker run commands?</p>
<p>We’re using buildx and the cache works great for Docker build and push workflows, but Docker runs are still a tad slow and we are running the same image in multiple places in our workflow. The ability to cache the image so it doesn’t have to pull down anything each time would be awesome…</p>

**Comment by AlexDCraig** [comment, marked as answer]
<p>The solution for this (that doesn’t really speed things up much, but was worth trying in my case) is to use Docker’s official build Action but use it with push false: and load: true along with the standard caching set. Then pull from cache each other place you are loading in this image.</p>

**Comment by Kurt-von-Laven** [comment]
<p>For folks who are only pulling and running Docker images, we just released <a href="https://github.com/marketplace/actions/docker-cache" rel="noopener nofollow ugc">docker-cache</a>, which caches all Docker images whether built or pulled. Please file an issue if you have any feedback!</p>
